src/components/publishing/sections/sections.tsx

```diff
@@ -34,10 +36,11 @@ const Sections: React.SFC<SectionsProps> = props => {
 function renderSections(article) {
   const sections = article.sections
   const renderedSections = sections.map((section, i) => {
-    const child = getSection(section)
+    const child = getSection(section, article.layout)
+    const layout = section.body ? findBlockquote(section.body, article.layout) : section.layout
```

Comment:
Hm this feels pretty specific to the text component -- would it be possible to move this logic there since section.body and article.layout is available in TextContainer?